pageserver/src/tenant/timeline.rs

```diff
@@ -1210,6 +1212,10 @@ impl Timeline {
             .start_timer();
         let mut results: BTreeMap<Key, Result<Bytes, PageReconstructError>> = BTreeMap::new();
         let layers_visited = reconstruct_state.get_layers_visited();
+        ctx.vectored_access_delta_file_cnt.fetch_add(
+            reconstruct_state.get_delta_layers_visited() as usize,
+            AtomicOrdering::SeqCst,
+        );
```

Comment:
This is a hot path. Doesn't feel right to add an atomic increment here which only compaction cares about.